cmd/machine-os-builder/start.go

```diff
@@ -95,11 +98,29 @@ func runStartCmd(_ *cobra.Command, _ []string) {
 		klog.Fatalln(err)
 	}
 
-	ctrl, err := getBuildController(ctx, cb)
-	if err != nil {
-		klog.Fatalln(err)
-	}
-
-	go ctrl.Run(ctx, 5)
-	<-ctx.Done()
+	leaderElectionCfg := common.GetLeaderElectionConfig(cb.GetBuilderConfig())
+
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Lock:            common.CreateResourceLock(cb, ctrlcommon.MCONamespace, componentName),
+		ReleaseOnCancel: true,
+		LeaseDuration:   leaderElectionCfg.LeaseDuration.Duration,
+		RenewDeadline:   leaderElectionCfg.RenewDeadline.Duration,
+		RetryPeriod:     leaderElectionCfg.RetryPeriod.Duration,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {
+				go common.SignalHandler(cancel)
+				ctrl, err := getBuildController(ctx, cb)
+				if err != nil {
+					klog.Fatalln(err)
+				}
+
+				go ctrl.Run(ctx, 5)
+				<-ctx.Done()
+			},
+			OnStoppedLeading: func() {
+				klog.Info("Stopped leading. Terminating.")
```

Comment:
Looks Good. Maybe for clarity modify this message? I know the MCO pod uses this same verbiage but not that we have a few components doing this, might make sense.